packages/apps/plugins/plugin-space/src/components/SpacePresence.tsx

```diff
@@ -46,44 +47,33 @@ export const SpacePresence = ({ object, spaceKey }: { object: TypedObject; space
     return null;
   }
 
-  // TODO(wittjosiah): This isn't working because of issue w/ deepsignal state.
-  //  Assigning plugin to deepsignal seems to create a new instance of objects in provides and breaks the reference.
-  const viewers = spacePlugin.provides.space.viewers
-    .filter((viewer) => {
-      return (
-        space.key.equals(viewer.spaceKey) && object.id === viewer.objectId && Date.now() - viewer.lastSeen < 30_000
-      );
-    })
-    .reduce(
-      (viewers, viewer) => {
-        viewers.set(viewer.identityKey, viewer.lastSeen);
-        return viewers;
-      },
-      new ComplexMap<PublicKey, number>((key) => key.toHex()),
-    );
+  const currentObjectViewers = spacePlugin.provides.space.viewersByObject.get(object.id) ?? defaultViewers;
```

Comment:
what if this component just returns `null` if this is undefined? this `defaultViewers` map I find confusing because it's hard to tell if anything actually gets stored in it

Reply:
Will adjust to check as part of the early null return.

Reply:
e2e fails if this returns `null` early, probably because it doesn’t touch the deepsignal. Will leave as-is for now.

Reply:
maybe leave a note as to what's going on then with that `defaultViewers` map basically being there to satisfy tsc